### PR TITLE
feat: cover null-typed bind parameter columns

### PR DESCRIPTION
**Pull request by fornwall**
Add a test that binds a batch whose parameter column has the Arrow type null.

Verified that existing drivers for sqlite, postgres, mysql, duckdb and flightsql passes.

Context: Had a bug/confusion around this myself when implementing a new driver, so this test would have helped.

### CODE REVIEW

File: adbc_drivers_validation/tests/statement.py [hunk 1]
```diff
@@ -121,6 +121,38 @@ def test_parameter_execute(
...
+            {
+                "0": pyarrow.array([7001], type=pyarrow.int64()),
+                "1": pyarrow.nulls(1),
+            }
+        )
+        with conn.cursor() as cursor:
+            cursor.adbc_statement.set_sql_query(
+                f"INSERT INTO {sample_table} ({id_}, {value}) "
+                f"VALUES ({driver.bind_parameter(1)}, {driver.bind_parameter(2)})"
+            )
```
Comment (lidavidm) [inline adbc_drivers_validation/tests/statement.py:143]: Shouldn't these queries be overridable?

File: adbc_drivers_validation/tests/statement.py [hunk 2]
```diff
@@ -121,6 +121,38 @@ def test_parameter_execute(
             result = pyarrow.RecordBatchReader._import_from_c(handle.address).read_all()
         assert result[0].to_pylist() == [2, 3, 4, 5]
 
+    @pytest.mark.requires_features(["statement_bind"])
```
Comment (lidavidm) [inline adbc_drivers_validation/tests/statement.py]: This also uses prepare, albeit bind _probably_ implies prepare already

File: adbc_drivers_validation/tests/statement.py [hunk 3]
```diff
@@ -121,6 +121,38 @@ def test_parameter_execute(
...
+        self,
+        driver: model.DriverQuirks,
+        conn: adbc_driver_manager.dbapi.Connection,
+        sample_table: str,
+    ) -> None:
+        id_ = driver.quote_identifier("id")
+        value = driver.quote_identifier("value")
+        parameters = pyarrow.RecordBatch.from_pydict(
+            {
+                "0": pyarrow.array([7001], type=pyarrow.int64()),
```
Comment (lidavidm) [inline adbc_drivers_validation/tests/statement.py:135]: Why do we need this column anyways?

Comment (fornwall) [reply adbc_drivers_validation/tests/statement.py:135]: I reused the `sample_table` test fixture (defined [here](https://github.com/fornwall/validation/blob/e363d8e3481e387480ff8f95f82a771d38fa5d5b/adbc_drivers_validation/tests/statement.py#L88)), instead of creating a new one. I think that makes sense, but let me know if you think otherwise!

File: adbc_drivers_validation/tests/statement.py (hunk 2 above)
Comment (fornwall) [reply adbc_drivers_validation/tests/statement.py]: Right, no drawback with being explicit? Added a dependency on `statement_prepare`: 70de9813abfebe7cd4a942b8c52407b55fc3a102

File: adbc_drivers_validation/tests/statement.py (hunk 1 above)
Comment (fornwall) [reply adbc_drivers_validation/tests/statement.py:143]: I was thinking that queries should be overridable when there is a need (an upfront knowledge if the sql is not portable, or if a driver has found a need to override it), and not every query needs to be overridable.

Not every existing query is overridable.

But let me know if you think every new query should be overridable (or at least if this new one should), and I'll add it here!

Comment (lidavidm) [reply adbc_drivers_validation/tests/statement.py:143]: I'm also debating now if this shouldn't be a regular txtcase, but I suppose I don't want a table entry for NA in the docs.

**Review by lidavidm — APPROVED** [review]
